ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -209,18 +295,12 @@ broadcastNetwork registry btime numCoreNodes pInfo initRNG numSlots = do
   Internal auxiliary
 -------------------------------------------------------------------------------}
 
--- | Communication channel used for the Chain Sync protocol
-type ChainSyncChannel m hdr = Channel m (AnyMessage (ChainSync hdr (Point hdr)))
-
--- | Communication channel used for the Block Fetch protocol
-type BlockFetchChannel m blk = Channel m (AnyMessage (BlockFetch blk))
-
 -- | The communication channels from and to each node
 data NodeChan m blk hdr = NodeChan
```

Comment:
Is it possible to revert this change and to use the "demuxed" channels here (I don't know whether that's possible).
Now, it says the the ChainSync consumer and producer can receive/send any message, including a BlockFetch message (and vice versa). Maybe this won't be the case at run-time, but it would be so much nicer to have this at compile-time too.

Reply:
This was not possible, for that reason in tests I needed to add `AnyProtocolMessage`.  The tests still run using a form of `Id` codec: still the only serialisation / de-serialisation that happens is wrapping and unwrapping of some existential types, but we loose type safety since the channels needed to have the same type.